Automattic/wp-calypso
# SPT: preview into an iFrame element

### Changes proposed in this Pull Request
This PR injects starter page template blocks into an `iframe` in order to more accurately preview what the templates look like.

A user wants the starter page template previews to match what they will see when they apply the template selection and are moved into the editor. 

#### Related issues:
Fixes #39341.
Fixes #38779.
Related #39348, #39352,  #39355.

### Dev Setup For Testing
See existing documentation for setting up FSE plugin development. In addition, you will need these plugins active on your local environment, likely via a mounted volume:
- Gutenberg
- Jetpack (with WPCOM connection or running in Development Mode)
- CoBlocks: https://wordpress.org/plugins/coblocks/
- Block Experiments: https://github.com/automattic/block-experiments

* Compile SPT plugin from `apps/full-site-editing` directory in calypso:
```cli
npx lerna run dev --scope='@automattic/full-site-editing' --stream --no-prefix
```
* Have WPCOM sandbox running


### Testing instructions

Once the env dev is set up:

1) Create a new page. The templates selector should appear.
2) Switch the template. See the template in the Large Preview.

_WIP..._

### TODO

We need to test each starter template to see if the preview matches the editor view. (select the template, then apply the layout by selecting the top right "Use ___ Layout" button). If one of them is broken, please document:
- which template it is, 
- which block is not rendering correctly, 
- and which theme you are using.
- if this error is also in `master` or not.
- if you consider this error blocking or not

**Template Sections to Check**
- [x] Blog
- [x] Contact
- [x] Portfolio
- [x] Team
- [ ] Home Pages

**Other**
- [x] Identify other bugs related to Starter Template preview that are not present in `master`
- [x] Fixing any identified bugs :)
- [x] `core/cover` block previewing
- [ ] Post Title preview should respect theme settings for alignment (centered vs left aligned, etc)
